### DISCUSSION TITLE
Getting Error failed to load USD file while running android project

### DISCUSSION THREAD

**Discussion by HetalDave2308** [syoyo/tinyusdz#105] [Q&A] [answered]
Hi, I have downloaded the zip file of this repository and I am trying to run the app in android studio. I am able to run the project, but the USD file is not displayed. Getting error "Failed to load USD file". I am running the project in arm64-8a emulator only. Can you please guide me if I am missing something?
Attaching screenshot for your reference.
![Screenshot_20231124_193929](https://github.com/syoyo/tinyusdz/assets/151940168/819c1a0e-ab67-4281-98c3-41cedd8d4df0)

**Comment by syoyo** [comment, marked as answer]
> Write iOS and Android example(2023 Winter expected)

in README.md.

iOS and Android example is TODO. The library should compile fine on mobile target, but you need to write your own code to display content for a while.
Contribution of Android example viewer is much appreciated!

**Reply by HetalDave2308** [reply]
Hi @syoyo, Thanks for the reply. I need your help to understand few things before implementing android example:
1. From `MainActivity.kt` `initScene()' is called from c++ file, which further calls `LoadUSDCFromFile()` followed by `LoadUSDCFromMemory()` and `USDCReader`. But the `initScene()` returns value 0 or -1, so can you please help me understand where exactly the USD file is parsed?
2. In order to display this USD we need a callback to android side and then we can display it in android component. So from where we can initiate that callback?

**Reply by syoyo** [reply]
You are first better to take a look at openglviewer example code: https://github.com/syoyo/tinyusdz/tree/dev/examples/openglviewer
